pkg/minikube/machine/start.go

```diff
@@ -70,15 +70,15 @@ func StartHost(api libmachine.API, cfg *config.ClusterConfig, n *config.Node) (*
 	machineName := driver.MachineName(*cfg, *n)
 
 	// Prevent machine-driver boot races, as well as our own certificate race
-	releaser, err := acquireMachinesLock(machineName)
-	if err != nil {
-		return nil, false, errors.Wrap(err, "boot lock")
-	}
+	releaser, err := acquireMachinesLock(machineName, cfg.Driver)
 	start := time.Now()
 	defer func() {
 		glog.Infof("releasing machines lock for %q, held for %s", machineName, time.Since(start))
 		releaser.Release()
 	}()
+	if err != nil {
+		return nil, false, errors.Wrap(err, "boot lock")
```

Comment:
Can you confirm this behaves as intended? It's unusual to release a lock on error, because you don't know if the lock was successfully acquired. It may cause a panic.

Reply:
I added a recover() to take care of that